### PR TITLE
Application Type clarifications

### PR DESCRIPTION
**Pull request by mmaddex**
This PR is intended to help make the requirements to enable third-party applications more discoverable, and to clarify that these settings are required for dynamically registered clients due to the fact that only third-party applications can be dynamically registered.

#### Modified pages
- https://docs-content-staging-pr-7068.herokuapp.com/docs/applications/application-types
- https://docs-content-staging-pr-7068.herokuapp.com/docs/api-auth/dynamic-client-registration
https://docs-content-staging-pr-7068.herokuapp.com/docs/applications/application-grant-types
#### New page
- https://docs-content-staging-pr-7068.herokuapp.com/docs/applications/enable-third-party-applications

### CODE REVIEW

File: articles/applications/enable-third-party-applications.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# Enabling third-party applications
+
+To enable [third-party applications](/application-types#third-party-applications) for your Tenant you must:
+
+- update your application's login page (if you use [Lock](/libraries/lock/v11))
+- promote the connections you will use with your third-party applications to **domain connections**
+
+## Update the login page
+
+To use the Auth0's [Universal Login](/hosted-pages/login) with third-party Applications, you must:
```
Comment (nicosabena) [inline articles/applications/enable-third-party-applications.md]: All of these only apply if Lock is used
```suggestion
If you use Lock in the [Hosted Login Page](/hosted-pages/login) you will need to:
```

File: articles/applications/enable-third-party-applications.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+- use at least version `10.7.x` of Lock
+- set `__useTenantInfo: config.isThirdPartyClient` when instantiating Lock
+- [PSaaS Appliance](/appliance) users must use `https://{config.auth0Domain}/` as the value for [the `configurationBaseUrl` option](https://github.com/auth0/lock#other-options) when instantiating Lock
+
+Sample script:
+
+```html
+<script src="https://cdn.auth0.com/js/lock/10.7/lock.min.js"></script>
```
Comment (nicosabena) [inline articles/applications/enable-third-party-applications.md]: Even though 10.7 is the minimum recommended, I would go for v11 in the example.
```suggestion
 <script src="https://cdn.auth0.com/js/lock/11.x.y/lock.min.js"></script>
```

File: articles/api-auth/dynamic-client-registration.md [hunk 3]
```diff
@@ -19,21 +19,15 @@ Dynamic Client Registration enables you to register [third-party applications](/
 
 This feature is based on the [OpenID Connect Dynamic Client Registration specification](https://openid.net/specs/openid-connect-registration-1_0.html) and in this article we will see how you can enable and use it.
 
-## Enable dynamic registration
-
-By default, the feature is disabled for all tenants. To change this, you have to:
+<%= include('../_includes/_enable-third-party-apps-info') %>
```
Comment (nicosabena) [inline articles/api-auth/dynamic-client-registration.md:22]: I don't see the include working in Heroku,not sure if that's expected.

Comment (mmaddex) [reply articles/api-auth/dynamic-client-registration.md:22]: fixed

**Review by khorne3 — APPROVED** [review]

**Pull request merged**